tests/unit/Exports/OrderExportTest.php

```diff
@@ -608,6 +609,107 @@ public function testOrderCreateWithEmballage()
         $this->processNewOrder($newOrderId, $newOrder);
     }
 
+    public function testCancelledOrder()
+    {
+        $this->initApiConnection();
+        $this->emptyEnvironment();
+
+        $orderHandler = new OrderHandler();
+
+        $syncedCancelledOrder = WC_Helper_Order::create_order();
+        $syncedCancelledOrder->update_status(OrderExport::STATUS_CANCELLED);
+        $syncedCancelledOrder->save();
+        $orderHandler->addMetadata($syncedCancelledOrder->get_id(), $syncedCancelledOrder);
+
+        $newToCancelledOrder = WC_Helper_Order::create_order();
+        $newToCancelledOrder->save();
+        $orderHandler->addMetadata($newToCancelledOrder->get_id(), $newToCancelledOrder);
+
+        $unsyncedCancelledOrder = WC_Helper_Order::create_order();
+        $unsyncedCancelledOrder->update_status(OrderExport::STATUS_CANCELLED);
+        $unsyncedCancelledOrder->save();
+        $orderHandler->addMetadata($unsyncedCancelledOrder->get_id(), $unsyncedCancelledOrder);
+
+        $unsyncedCancelledOrderTaskIds = TaskModel::getTasksByStoreKeeperId($unsyncedCancelledOrder->get_id());
+        // Simulate marking tasks as success
+        foreach ($unsyncedCancelledOrderTaskIds as $taskId) {
+            TaskModel::update($taskId, ['status' => TaskHandler::STATUS_SUCCESS]);
+        }
+
+        $sentOrder = [];
+        StoreKeeperApi::$mockAdapter
+            ->withModule(
+                'ShopModule',
+                function (MockInterface $module) use (&$sentOrder, $syncedCancelledOrder) {
+                    $module->allows('findShopCustomerBySubuserEmail')->andReturnUsing(
+                        function () {
+                            return ['id' => mt_rand()];
+                        }
+                    );
+
+                    $module->allows('naturalSearchShopFlatProductForHooks')->andReturnUsing(
+                        function () {
+                            return [
+                                'data' => [],
+                                'total' => 0,
+                                'count' => 0,
+                            ];
+                        }
+                    );
+
+                    $module->allows('newOrder')->andReturnUsing(
+                        function ($params) use (&$sentOrder, $syncedCancelledOrder) {
+                            [$order] = $params;
```

Comment:
how do we know it was not synchronized? 

Maybe adding `throw new Exception('Shall not sync!');` here will make it clear it's never called

Reply:
Added exception throwing.

Reply:
```
if ($order['shop_order_number'] === $unsyncedCancelledOrder) 
```
I don't think it really throws anything `string !== object` , can you debug over it to make sure?

Reply:
My bad. Should be `$unsyncedCancelledOrder->get_id()`.